hnn_core/gui/gui.py

```diff
@@ -504,167 +922,199 @@ def load_drives(variables, params, log_out, drives_out, drive_widgets,
             )
 
 
-def on_upload_change(change, sliders, params, tstop, tstep, log_out, variables,
-                     drive_boxes, drive_widgets, drives_out):
+def on_upload_change(change, params, tstop, dt, log_out, simulation_data,
+                     drive_boxes, drive_widgets, drives_out, connectivity_out,
+                     connectivity_sliders, load_info):
     if len(change['owner'].value) == 0:
         return
 
-    params_fname = change['owner'].metadata[0]['name']
+    # params_fname = change['owner'].metadata[0]['name']
+    params_fname = list(change['owner'].value.keys())[0]
     file_uploaded = change['owner'].value
     param_data = list(file_uploaded.values())[0]['content']
     param_data = codecs.decode(param_data, encoding="utf-8")
 
+    if load_info['prev_param_data'] == param_data:
+        with log_out:
+            print("Same param. No reloading. \
```

Comment:
```suggestion
            print("Same param. No reloading."
```
nitpick ... backslashes are not needed within parenthesis. They make the code look ugly. Better to use parenthesis and align.